gen3/bin/kube-setup-workvm.sh

```diff
@@ -165,12 +165,39 @@ EOM
       sudo ln -s /usr/local/bin/aws-iam-authenticator heptio-authenticator-aws
     )
   fi
-  if ! which helm > /dev/null 2>&1; then
-    helm_release_URL="https://storage.googleapis.com/kubernetes-helm/helm-v2.11.0-linux-amd64.tar.gz"
-    curl -o "${XDG_RUNTIME_DIR}/helm.tar.gz" ${helm_release_URL}
-    tar xf "${XDG_RUNTIME_DIR}/helm.tar.gz" -C ${XDG_RUNTIME_DIR}
-    sudo mv "${XDG_RUNTIME_DIR}/linux-amd64/helm" /usr/local/bin
-  fi
+  ( # in a subshell install helm
+    install_helm() {
+      helm_release_URL="https://get.helm.sh/helm-v3.3.0-linux-amd64.tar.gz"
+      curl -o "${XDG_RUNTIME_DIR}/helm.tar.gz" ${helm_release_URL}
+      tar xf "${XDG_RUNTIME_DIR}/helm.tar.gz" -C ${XDG_RUNTIME_DIR}
+      sudo mv -f "${XDG_RUNTIME_DIR}/linux-amd64/helm" /usr/local/bin
+
+      # helm3 has no default repo, need to add it manually
+      helm repo add stable https://kubernetes-charts.storage.googleapis.com
+      helm repo update
+    }
+
+    migrate_helm() {
+      if ! helm plugin list | grep 2to3 > /dev/null 2>&1; then
+        helm plugin install https://github.com/helm/helm-2to3.git
+      fi
+      helm 2to3 convert grafana
+      helm 2to3 convert prometheus
+      
+      # delete tiller and other helm2 data/configs
+      helm 2to3 cleanup --skip-confirmation
+    }
+
+    if ! which helm > /dev/null 2>&1; then
+      install_helm
+    else 
+      # Overwrite helm2 with helm3
+      if ! helm version --short | grep v3 > /dev/null 2>&1; then
```

Comment:
It's a little tricky.  We can have multiple VPC's administered by different user's in the same VM.  So it's possible for something like this to happen:
* vpc1 updates - helm gets updated, and prometheus and grafana get migrated
* vpc2 updated - help has already been updated, so prometheus and grafana don't migrate

You may want to move the migrate out to `kube-setup-tiller` - I assume that doesn't require `sudo` too (most of our accounts don't have sudo), and we want to make sure the migrate happens before we drop tiller probably?

Reply:
This step is not very idempotent, so I am afraid if we add it to `kube-setup-tiller` we will end up erroring out if the migration has already happened. I need to make it more idempotent somehow, where the migrations only happens if these conditions are met: 

- Client is helm3

- Tiller is still deployed

- Grafana and prometheus deployments exist, but not as a helm release.